packages/frontend/app/services/channels/vote-sorter.ts

```diff
@@ -0,0 +1,29 @@
+import Service from '@ember/service';
+import VoteChain from 'emberclear/models/vote-chain';
+import Identity from 'emberclear/models/identity';
+
+export default class VoteSorter extends Service {
```

Comment:
Because this service doesn't manage any state, I think its contents could be migrated to helper functions that would just be used by the VoteChains

Reply:
The purpose of having it here was that the action of creating a vote will also need to use this in order to create new votes. That being said, it may be a thing where we can have a single vote manager service that handle creation and verification.

Reply:
but a service manages shared state in the app.

if there is no state to manage, and it's just a function that receives input and returns some output, this could be sufficient:

```ts
export function nameOfFunction(inputs) {
  return theTheing;
}
```
and then you can 
```ts
import { nameOfFunction } from 'path/to/the-function';
```

but if there is a plan to have persistent state here later, then ignore me :D

Reply:
In that case, it could be moved elsewhere. Would it still go in the services package then?

Reply:
Since we are relying on the WorkersService, I think it needs to stay a service technically even though it isn't really holding state. The WorkersService is how we can use the crypto needed for the verification.

Reply:
I,m not sure if that makes sense? As is, these functions can exist in module-space, and anyone can import them. No D.I. needed ;)

Reply:
> Would it still go in the services package then?

There are a couple other examples of utility modules in the services sub folders. You could use those for examples?

Reply:
Given that they could just be imported as functions, how would we use the WorkersService? Would users of the function have to inject it themselves?

Reply:
it's pretty much the same, but conveys better semantics.

instead of `@service('channels/vote-sorter') sorter;`

you'd do 
```ts
import { generateSortedVote as toSortedVoteChain } from 'wherever-the-module-lives';
```

Reply:
idk if this seems like toemaytoe tomahtoe, but services have more of a memory and initial runtime cost to them than modules do.

Reply:
> how would we use the WorkersService? 

how you're using the worker service now is  fine. `VoteSorter` doesn't inject anything -- it has no dependencies.

Reply:
Makes sense. Fixed that.